src/interfaces/chain.h

```diff
@@ -131,6 +131,10 @@ class Chain
 public:
     virtual ~Chain() = default;
 
+    // Removes the transaction from the mempool and returns true if it was
+    // removed. Returns false if the transaction was not in the mempool.
```

Comment:
What happens if the mempool tx has dependent transactions? (Other txs that spend its outputs) Do all transactions get removed or does this function return `false`?

Answered my own question below, add a docstring about this here.